src/charm.py

```diff
@@ -103,7 +125,123 @@ def _on_stop(self, _: ops.StopEvent) -> None:
     def _on_upgrade_charm(self, _: ops.UpgradeCharmEvent) -> None:
         """Handle upgrade-charm."""
         self.unit.status = ops.MaintenanceStatus("Upgrading dependencies")
-        self.bind.prepare(self.unit.name)
+        self.bind.setup(self.unit.name)
+
+    def _on_leader_elected(self, _: ops.LeaderElectedEvent) -> None:
+        """Handle leader-elected event."""
+        # We check that we are still the leader when starting to process this event
+        if self.unit.is_leader():
```

Comment:
Should a non-leader unit do something once a new leader is elected?reload_bind?

Reply:
In this PR it doesn't need to do anything since it has the latest config already running. But in the next PR, it will need to "activate" from a standby mode.